pandas/core/arrays/timedeltas.py

```diff
@@ -227,6 +242,18 @@ def _validate_fill_value(self, fill_value):
                              "Got '{got}'.".format(got=fill_value))
         return fill_value
 
+    @property
+    def is_monotonic_increasing(self):
```

Comment:
Where is this used in this PR?

Reply:
When calling the constructor with `freq="infer"` we end up calling `tseries.frequencies.infer_freq`, which accesses this property

Reply:
I think I already expressed this previously, but I would prefer that we have this discussion for all the Arrays, not just for TimedeltaArray (or DatetimeArray), as this is not a datetime-specific attribute. If you don't want to have that discussion first, you can always make it a private attribute here, and check that in `infer_freq` (or call the `algos` methods there if the passed values don't have such an attribute)

Reply:
If this is a discussion you'd like to have for EAs more generally, go ahead and open an issue for it.

I would be +1 on putting these attributes in the mixin class so that they are available on all three of DTA/TDA/PA, but for now they are only needed on TDA.

Special-casing inside `infer_freq` (or more specifically `FrequencyInferer` is a pretty ugly hack that I'd rather avoid.  That said, if you're going to insist on it, I might as well get it over with.

Reply:
@jbrockmendel I would really prefer if you leave this out of this PR (I mean adding the attributes, so which means special casing this in `infer_freq`). 

Looking at the code again, I think the easiest to do is to ensure what is passed to `_FrequencyInferrer` is an actual index. So if we have an array, simply convert to Index without copy.

Reply:
converting to an Index would break the "Array should be ignorant of Index" rule discussed elsewhere.

I'll special-case this to get this over with, but I maintain this is introducing a code smell.